test/keymgmt_internal_test.c

```diff
@@ -88,7 +88,7 @@ static int get_ulong_via_BN(const OSSL_PARAM *p, unsigned long *goal)
     int ret = 1;                 /* Ever so hopeful */
 
     if (!TEST_true(OSSL_PARAM_get_BN(p, &n))
-        || !TEST_true(BN_bn2nativepad(n, (unsigned char *)goal, sizeof(*goal))))
+        || !TEST_int_ge(BN_bn2nativepad(n, (unsigned char *)goal, sizeof(*goal)), 0))
```

Comment:
I think this should be `TEST_int_gt()`.  A zero return on failure is being accepted as written.

Reply:
But all other invoking places use < 0 as an error and >=0 as success. 

See 
 + crypto/params.c:940
 + crypto/evp/ctrl_params_translate.c:468
 + crypto/evp/p_lib.c:2271
 + test/params_test.c:337
 + and so on

Reply:
Yeah, you are correct.